home/tests/unit/dailywalk/test_get.py

```diff
@@ -0,0 +1,187 @@
+        self.assertEqual(response.status_code, 200)
+        # Parse the response
+        response_data = response.json()
+        fail_message = f"Server response - {response_data}"
+        self.assertEqual(response_data["status"], "success", msg=fail_message)
+        self.assertIn("daily_walks", response_data, msg=fail_message)
+        self.assertIn("total_steps", response_data, msg=fail_message)
+        # Check if total steps is correct
+        self.assertEqual(response_data["total_steps"], sum([dw["steps"] for dw in self.daily_walks]), msg=fail_message)
+        # Check if the number of events match
+        self.assertEqual(len(response_data["daily_walks"]), len(self.daily_walks), msg=fail_message)
+        for walk in response_data["daily_walks"]:
+            self.assertIn("date", walk, msg=fail_message)
+            self.assertIn("steps", walk, msg=fail_message)
+        # Check if they have the same exact data
+        # Note that the response will always be ordered by the latest date
+        self.assertEqual(
+            response_data["daily_walks"],
+            sorted(
+                [{"date": dw["date"], "steps": dw["steps"]} for dw in self.daily_walks],
+                key=lambda x: x["date"],
+                reverse=True,
+            ),
+            msg=fail_message,
+        )
+
+    def test_dailwalk_get_withparam_reverse(self):
+        self.request_params["reverse_order"] = True
+        response = self.client.post(path=self.url, data=self.request_params, content_type=self.content_type)
+        # Check for a successful response by the server
+        self.assertEqual(response.status_code, 200)
+        # Parse the response
+        response_data = response.json()
+        fail_message = f"Server response - {response_data}"
+        self.assertEqual(response_data["status"], "success", msg=fail_message)
+        self.assertIn("daily_walks", response_data, msg=fail_message)
+        self.assertIn("total_steps", response_data, msg=fail_message)
+        # Check if total steps is correct
+        self.assertEqual(response_data["total_steps"], sum([dw["steps"] for dw in self.daily_walks]), msg=fail_message)
+        # Check if the number of events match
+        self.assertEqual(len(response_data["daily_walks"]), len(self.daily_walks), msg=fail_message)
+        for walk in response_data["daily_walks"]:
+            self.assertIn("date", walk, msg=fail_message)
+            self.assertIn("steps", walk, msg=fail_message)
+        # Check if they have the same exact data
+        # Note that the response will now be ordered by the first date
+        # So reverse=True for sort is removed
+        self.assertEqual(
+            sorted(response_data["daily_walks"], key=lambda x: x["date"]),
+            sorted([{"date": dw["date"], "steps": dw["steps"]} for dw in self.daily_walks], key=lambda x: x["date"]),
+            msg=fail_message,
+        )
+
+    def test_dailwalk_get_withparam_numwalks(self):
```

Comment:
Is this for pagination? How does one query for the next page?

Reply:
Initially, I added this in since I thought the use case was the phone displaying the most recent walks. I wasn't fully sure how things would work on the app side. Now it seems like since the phone has all the data anyway, this parameter isn't super useful or even needed. So I'll likely remove all this and simplify it. We wont need pagination since the app can call and get everything at once and store it if needed